### PR TITLE
Quieter datum warning

### LINKED ISSUE #5749: Improve datum FUTURE warning
## 🐛 Bug Report

When loading any netCDF file, a warning is produced from iris:

> FutureWarning: Ignoring a datum in netCDF load for consistency with existing behaviour. In a future version of Iris, this datum will be applied. To apply the datum when loading, use the iris.FUTURE.datum_support flag.

The warning text implies that there is a datum in the file being loaded and it is being ignored.  Can the behaviour to trigger the warning be aligned with the warning text, so the warning only fires if there is a datum in the source file?

This would make the warning much more useful for end users: users would only see the warning for those cases where there is a behaviour change controlled by the iris.FUTURE.datum_support flag (and hence where the user may need to understand or do something).  At the moment, the warning is emitted for files where there will be no behaviour change too.

## How To Reproduce
Steps to reproduce the behaviour:

1. Load a netCDF file with iris 3.7 (or iris since 3.3ish?)

## Expected behaviour

Only seeing the warning for netCDF files that do have a datum defined.

### PR DESCRIPTION
**Pull request by trexfeathers**
### Description

Closes #5749 

---
[Consult Iris pull request check list]( https://scitools-iris.readthedocs.io/en/latest/developers_guide/contributing_pull_request_checklist.html)

---
Add any of the below labels to trigger actions on this PR:

- https://github.com/SciTools/iris/labels/benchmark_this

### CODE REVIEW

File: docs/src/whatsnew/latest.rst [hunk 1]
```diff
@@ -53,6 +53,10 @@ This document explains the changes made to Iris for this release
 #. `@pp-mo`_ corrected the use of mesh dimensions when saving with multiple
    meshes.  (:issue:`5908`, :pull:`6004`)
 
+#. `@trexfeathers`_ fixed the datum :class:`python:FutureWarning` to only be raised if
+   the ``datum_support`` :class:`~iris.Future` flag is disabled AND a datum is
+   present the loaded NetCDF grid mapping. (:issue:`5749`, :pull:`6050`)
```
Comment (HGWright) [inline docs/src/whatsnew/latest.rst:58]: "AND a datum is present the loaded NetCDF grid mapping." I think this should be "AND a datum is present (ON) the loaded NetCDF grid mapping."

**Review by HGWright — CHANGES_REQUESTED** [review]
Thanks @trexfeathers, just a small comment about the whatsnew entry grammar.

File: docs/src/whatsnew/latest.rst (hunk 1 above)
Comment (trexfeathers) [reply docs/src/whatsnew/latest.rst:58]:
```suggestion
   present on the loaded NetCDF grid mapping. (:issue:`5749`, :pull:`6050`)
```

**Review by HGWright — APPROVED** [review]
Thanks for that change.

**Pull request merged**